### PR TITLE
Recognise and display indicators of compromise

### PR DESCRIPTION
**Pull request by msariego-endgame**

### CODE REVIEW

**Comment by jecarr** [comment]
*Excuse my base-branch tampering - I noted some commits were from the previous PR but I think that's because I did a rebase rather than a merge (so the changes are coming up again)

**Review by jecarr — CHANGES_REQUESTED** [review]
Really liking the style of this, just a front-end tweak to do.

I did add two commits - pdf column widths and permission checks. The latter is a chunkier commit so feel free to contest how I did it/improve where you feel is best.

File: threadcomponents/handlers/web_api.py [hunk 1]
```diff
@@ -472,6 +478,18 @@ async def pdf_export(self, request):
...
+
+        # Table for indicators of compromise
+        ioc_header_row = []
+        for column_header in ['Indicators of Compromise']:
+            ioc_header_row.append(dict(text=column_header, style='bold'))
+        ioc_table = dict(widths=['100%'], body=[ioc_header_row])
+        for sentence in sentences:
+            if any(ioc['sentence_id'] == sentence['uid'] for ioc in indicators_of_compromise):
+                ioc_table['body'].append([sentence['text']])
+        dd['content'].append(dict(table=ioc_table))
```
[the remark this answers is not in the document]
Comment (KadeMorton) [reply threadcomponents/handlers/web_api.py:491]: @jecarr no that's fine.

File: webapp/theme/scripts/basics.js [hunk 2]
```diff
@@ -354,6 +357,10 @@ function updateSentenceContext(data) {
   $(missingTechBtn).prop("disabled", $(`.${highlightClassImg}`).length > 0 || !enableSenButtons);
   // Allow 'remove selected' button
   $(delSenBtn).prop("disabled", !enableSenButtons);
+  // Allow 'Indicator of Compromise' switch
+  $(iocSwitchSelector).prop("disabled", !enableSenButtons);
+  const sentenceIsIoc = $(`#elmt${sentence_id}`).attr("data-ioc") === "true";
+  $(iocSwitchSelector).prop("checked", sentenceIsIoc);
```
[the remark this answers is not in the document]
Comment (msariego-endgame) [reply webapp/theme/scripts/basics.js:363]: Oh, you're correct.
Thanks for noticing it!

**Review by jecarr — APPROVED** [review]

**Pull request merged**